MediaBrowser.Controller/Entities/BaseItemExtensions.cs

```diff
@@ -64,21 +64,27 @@ public static void SetImagePath(this BaseItem item, ImageType imageType, string
         where T : BaseItem
         where TU : BaseItem
         {
-            var sourceProps = typeof(T).GetProperties().Where(x => x.CanRead).ToList();
-            var destProps = typeof(TU).GetProperties()
-                    .Where(x => x.CanWrite)
-                    .ToList();
+            var destProps = typeof(TU).GetProperties().Where(x => x.CanWrite).ToList();
 
-            foreach (var sourceProp in sourceProps)
+            foreach (var sourceProp in typeof(T).GetProperties())
             {
-                if (destProps.Any(x => x.Name == sourceProp.Name))
+                if (!sourceProp.CanRead)
                 {
-                    var p = destProps.First(x => x.Name == sourceProp.Name);
-                    p.SetValue(dest, sourceProp.GetValue(source, null), null);
+                    continue;
                 }
 
-            }
+                var p = destProps.Find(x => x.Name == sourceProp.Name);
+                if (p == null)
+                {
+                    continue;
+                }
 
+                var v = sourceProp.GetValue(source, null);
```

Comment:
I would suggest moving this to before `destProps.Find()` call as it could make `continue` earlier :)

Reply:
This being `var v = sourceProp.GetValue(source, null);`?

Reply:
Yes, I suggest first getting the property value and look for destination property only if source property isn't `null`.